lib/authority.js

```diff
@@ -214,12 +221,19 @@ Authority.prototype._validateViaInstanceDiscovery = function(callback) {
  * @param {Authority.GetOauthEndpointsCallback}  callback  The callback function.
  */
 Authority.prototype._getOAuthEndpoints = function(tenantDiscoveryEndpoint, callback) {
-  if (this._tokenEndpoint) {
+  if (this._tokenEndpoint && this._deviceCodeEndpoint) {
     callback();
     return;
   } else {
     // fallback to the well known token endpoint path.
-    this._tokenEndpoint = url.format(this._url) + AADConstants.TOKEN_ENDPOINT_PATH;
+    if (!this._tokenEndpoint){
+       this._tokenEndpoint = url.format(this._url) + AADConstants.TOKEN_ENDPOINT_PATH;
+    }
+
+    if (!this._deviceCodeEndpoint){
+       this._deviceCodeEndpoint = url.format(this._url) + AADConstants.DEVICE_ENDPOINT_PATH;
```

Comment:
Hmmm, well I've checked both node 12.7 and 10.25. I'd be interested if your node installation doesn't add the slash. But I'm seeing

```
$ node
> url.format('http://contoso.org')
'http://contoso.org/'
>
```